egui_glow/src/painter.rs

```diff
@@ -581,29 +521,61 @@ impl Painter {
 
     /// This function must be called before Painter is dropped, as Painter has some OpenGL objects
     /// that should be deleted.
+    #[cfg(debug_assertions)]
     pub fn destroy(&mut self, gl: &glow::Context) {
-        debug_assert!(!self.destroyed, "Only destroy the egui glow painter once!");
-
+        assert!(!self.destroyed, "Only destroy once!");
         unsafe {
             self.destroy_gl(gl);
+            if let Some(ref post_process) = self.post_process {
+                post_process.destroy(gl);
+            }
         }
-
         self.destroyed = true;
     }
 
+    #[cfg(not(debug_assertions))]
+    pub fn destroy(&self, gl: &glow::Context) {
+        unsafe {
+            self.destroy_gl(gl);
+            if let Some(ref post_process) = self.post_process {
+                post_process.destroy(gl);
+            }
+        }
+    }
+
+    #[cfg(debug_assertions)]
     fn assert_not_destroyed(&self) {
-        debug_assert!(
-            !self.destroyed,
-            "the egui glow painter has already been destroyed!"
-        );
+        assert!(!self.destroyed, "egui has already been destroyed!");
     }
+
+    #[inline(always)]
+    #[cfg(not(debug_assertions))]
+    #[allow(clippy::unused_self)]
+    fn assert_not_destroyed(&self) {}
 }
+// ported from egui_web
+pub fn clear(gl: &glow::Context, dimension: [u32; 2], clear_color: egui::Rgba) {
+    unsafe {
+        gl.disable(glow::SCISSOR_TEST);
+
+        gl.viewport(0, 0, dimension[0] as i32, dimension[1] as i32);
 
+        let clear_color: Color32 = clear_color.into();
+        gl.clear_color(
+            clear_color[0] as f32 / 255.0,
+            clear_color[1] as f32 / 255.0,
+            clear_color[2] as f32 / 255.0,
+            clear_color[3] as f32 / 255.0,
+        );
+        gl.clear(glow::COLOR_BUFFER_BIT);
+    }
+}
 impl Drop for Painter {
     fn drop(&mut self) {
-        debug_assert!(
+        #[cfg(debug_assertions)]
+        assert!(
             self.destroyed,
-            "Make sure to call destroy() before dropping to avoid leaking OpenGL objects!"
```

Comment:
you seems to have accidentally reverted some cleanup like this that I made in https://github.com/emilk/egui/commit/8accfd9a8f2d21ada20661fda34e76913deef0b6

can you please try reverting these reverts?

Reply:
maybe i used old base revision .